pkg/drivers/kvm/numa.go

```diff
@@ -65,10 +66,13 @@ func numaXML(cpu, memory, numaCount int) (string, error) {
 		numaNodes[i].CPUCount++
 	}
 	for i := range numaNodes {
-		CPUTopologySlice := make([]string, 0)
-		for seq := CPUSeq; seq < CPUSeq+numaNodes[i].CPUCount; seq++ {
-			CPUTopologySlice = append(CPUTopologySlice, strconv.Itoa(seq))
-		}
+		CPUTopologySlice := slices.Collect(func(yield func(string) bool) {
+			for seq := CPUSeq; seq < CPUSeq+numaNodes[i].CPUCount; seq++ {
+				if !yield(strconv.Itoa(seq)) {
+					return
+				}
+			}
+		})
```

Comment:
Keep old code.

Unrelated to the slices.Collect change - here we know the size of the slice since we add all items, so we can use:

    make([]string, 0, numaNodes[i].CPUCount)